### ISSUE TITLE
Enumerate Silently Broken in UniqueToGameObjectMultitonPool and UniqueToEntityMultitonPool

### ISSUE THREAD

**Issue by DonLoquacious** [lordofduct/spacepuppy-unity-framework#8]
In MultitonPool.cs in Base/Collections, there's an Enumerate function that's supposed to assist in iterating through just a given sub-type in the pool. In the base version it uses a hashtable for the underlying collection type, but in the derived UniqueToGameObjectMultitonPool and UniqueToEntityMultitonPool classes, the underlying collection type is a dictionary instead, and it looks like the function was just copied over from the base.

So, in the Enumerate<TSub> functions for the derived types (lines ~589, ~1001), they check if e.Current is TSub, which it can never be because e.Current is a key/value pair, so these need to be changed to "e.Current.Value is TSub" I think, and of course return the Value instead of the key/value pair.

**Comment by lordofduct** [comment]
Should already be that.

Not sure if the release matches this, built new release.

**Comment by DonLoquacious** [comment]
I wasn't using a release, but the loose scripts in the repo, and it's definitely broken there. Check ~589 and ~1001 in SpacepuppyBase/Collections/MultitonPool.cs.

``public IEnumerable<TSub> Enumerate<TSub>() where TSub : class, T
{
    var e = _pool.GetEnumerator();
    while (e.MoveNext())
    {
        if (e.Current is TSub) yield return e.Current as TSub;
    }
}``

... in both derived classes needs to be something like...

``public IEnumerable<TSub> Enumerate<TSub>() where TSub : class, T
{
    var e = _pool.GetEnumerator();
    while (e.MoveNext())
    {
        if (e.Current.Value is TSub) yield return e.Current.Value as TSub;
    }
}``

... because e.Current refers to a KeyValuePair and not just a value, so it'll never be of type TSub.

Sorry if there's some sort of miscommunication happening here- just trying to help, and I had to fix this when I downloaded the repo earlier today. **shrugs**
